stdlib/cache/src/main/ballerina/src/cache/abstract_cache.bal

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents Ballerina `AbstractCache` object and cache related operations. Any custom cache implementation should be
+# object-wise similar.
+public type AbstractCache abstract object {
+
+    # Adds the given key, value pair to the cache.
+    #
+    # + key - Key of the cached value
+    # + value - Value to be cached
+    # + maxAgeInSeconds - The value in seconds which the cache entry is valid. '-1' means, the entry is valid forever.
+    public function put(string key, any value, int maxAgeInSeconds);
```

Comment:
Can we make maxAgeInSeconds an optional param with -1 as default value? 
IMO maxAgeInSeconds value will not be changed specifically for the 80% of the case.

Reply:
Yes. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/21308/commits/0cce9e3259d7d174a4cb0e0549dbed102a919d4a